src/System.Diagnostics.Process/src/System/Diagnostics/AsyncStreamReader.cs

```diff
@@ -280,10 +219,10 @@ private void GetLinesFromStringBuilder()
                 _currentLinePos = 0;
             }
 
-            FlushMessageQueue();
+            FlushMessageQueue(token);
         }
 
-        private void FlushMessageQueue()
+        private void FlushMessageQueue(CancellationToken token)
```

Comment:
Rather than passing the CancellationToken around everywhere, what would you think about passing it just into the ctor and storing it as a readonly instance field?